todo.js

```diff
@@ -42,6 +58,60 @@ function createDeleteButton(todoId) {
   return deleteButtonElement;
 }
 
+function createEditButton(todoId) {
+  const editButtonElement = document.createElement("button");
+  editButtonElement.innerHTML = "Edit";
+  editButtonElement.addEventListener("click", function () {
+    handleTodoEdit(todoId);
+  });
+  return editButtonElement;
+}
+
+function createUpdateButton(todoId) {
+  const updateButtonElement = document.createElement("button");
+  updateButtonElement.innerHTML = "Update";
+  updateButtonElement.addEventListener("click", () => handleTodoUpdate(todoId));
+  return updateButtonElement;
+}
+
+function createCancelButton(todoId) {
+  const cancelButtonElement = document.createElement("button");
+  cancelButtonElement.innerHTML = "Cancel";
+  cancelButtonElement.addEventListener("click", () => cancelEdit(todoId));
+  return cancelButtonElement;
+}
+
+function handleTodoEdit(todoId) {
+  todos.forEach((todo) => {
+    if (todo.id === todoId) {
+      todo.editState = true;
+    } else {
+      todo.editState = false;
+    }
+  });
+  renderTodoList();
+}
```

Comment:
Let's use early returns here instead of `if..else`

```suggestion
function handleTodoEdit(todoId) {
  todos.forEach((todo) => {
    if (todo.id === todoId) {
      todo.editState = true;
      
      return
    }
    
    todo.editState = false;
  });
  renderTodoList();
}
```